### ISSUE TITLE
Allow switching to png rendering

### ISSUE THREAD

**Issue by YPetremann** [sverweij/atom-graphviz-preview-plus#14]
I'm fond of graphviz because it allow to make really cool diagrams and with your plugin, it's possible to view them as you code.

But here's a problem, with viz.js, it's almost impossible to use images correctly because it doesn't seems to detect the image, then with Graphviz Command Line, the only way I've found to use images is to put a absolute path (which I don't want) and finaly when using imagepath attr, dot compilation is fine but the images can't be shown because they are missing, this because the rendered image is in svg.

So for people that need to use images in the graph, it should be interesting to allow switching to png rendering when Graphviz Command Line is enabled.

**Comment by sverweij** [comment]
Interesting suggestion. I'll give it a spin and will look into whether I can make it so it addresses your issue (actually showing the images); I recall some hairiness from my last encounter with external images in GraphViz.

**Comment by sverweij** [comment]
There's two things that cause images in graphviz sources to not work well, both have to do with paths and both have to be solved:
1. When graphviz-preview-plus spawns `dot` it doesn't pass a working directory - so `dot` gets the default (`/` on macOS - maybe something else on other OS's). `dot` does not look for image files relative to the file.
fix: pass a working directory when spawning dot. I think using Atom's current working directory is probably most in line with how dot works  (_does that sound reasonable?_), 
2. After `dot` successfully rendered the svg gvpp passes it on to atom for rendering. Atom has its own concepts of where to find pictures. If the image has no absolute path in there it assumes the picture to reside in `%atomhome%/Content/Resources/app/static` - which usually means it shows the 'no image found' image. Here your png rendering solution will probably be the 

(This means that in the current live version of graphviz-preview-plus images will _only_ work when we use absolute paths in the source (see below).
```dot
digraph  {
  bgcolor=transparent
  edge[color=grey]
  node[shape=none image="/full/path/to/picture.png"]
  a -> b -> c -> a
}
```
![screen shot 2017-06-09 at 20 09 48](https://user-images.githubusercontent.com/4822597/26988433-ab1599ac-4d4f-11e7-832f-b70c789c16d1.png)

**Comment by YPetremann** [comment]
I don't know what's possible in atom, but maybe have the possibility to execute dot while in project folder or in gv file folder would be a great addition.

**Comment by sverweij** [comment]
I'm working on that. 

It's possible to pass a working directory to (the child process that spawns) dot. I just have to find a way to (reliably) find path from the atom env that _(1) reflects the current project path_ (ideal) or if it isn't there _(2) reflects the gv file's folder_. 

(2) is doable. I'm hunting in the atom docs & source to get (1) done, though.

**Comment by sverweij** [comment]
Little update:
- :white_check_mark: g-p-p now passes the files folder to dot, so dot now generates the correct svg (including pictures) 
- I have solution for the second problem (convincing atom it should find the images in the neighbourhood of the file instead of on the _resourcePath_) as well. The good news is that g-p-p can stay with using svg's for this. 
  - markdown preview faces the same problem, yet shows images. I'm currently stealing & modifying their solution for use with svg's. 
  - After that I'll need to change something in the implementation of the _copy to clipboard_ and _save as svg_ features so they keep working as they currently do.

I'll have to release these things in concert so I don't expect a release with these changes in this week.

**Comment by sverweij** [comment]
That was easier than anticipated. Needs some polishing, but so far ...

![something-works](https://user-images.githubusercontent.com/4822597/27353624-f6f51a74-5604-11e7-930f-9278746bf21c.gif)

**Comment by sverweij** [comment]
@YPetremann I've published version v1.4.3 of graphviz-preview-plus to atom.io that renders images embedded into .dot sources.
- If the source is a file, it will seek the images relative to that file
- If the source is a buffer, it will seek the images relative to the (first) working directory of the current project (... and when there's no project it will use whatever is the default working directory for atom contexts - usually '/')

Let me know how it work for you - and whether it solves your issue.

**Comment by sverweij** [comment]
Closing, assuming v1.4.3 solves the underlying problem.
